### PR TITLE
Document self-updating feature for ConfigMap

### LINKED ISSUE #1407: user-guide/configmap does not document self-updating
I can't find anything that documents the ability to have configmap volumes self-updated.

Secrets documents it with two sentences: "When a secret being already consumed in a volume is updated, projected keys are eventually updated as well. The update time depends on the kubelet syncing period."

@pmorie

### PR DESCRIPTION
**Pull request by wojtek-t**
Fix #1407

Ref https://github.com/kubernetes/website/pull/21189#issuecomment-636360446

/assign @sftim

### CODE REVIEW

File: content/en/docs/concepts/configuration/configmap.md [hunk 1]
```diff
@@ -157,6 +157,91 @@ or {{< glossary_tooltip text="operators" term_id="operator-pattern" >}} that
 adjust their behavior based on a ConfigMap.
 {{< /note >}}
 
+## Using ConfigMaps
+
+ConfigMaps can be mounted as data volumes. ConfigMaps can also be used by other
+parts of the system, without being directly exposed to the Pod. For example,
+ConfigMaps can hold data that other parts of the system should use for configuration.
+
+### Using Secrets as files from a Pod
```
Comment (sftim) [inline content/en/docs/concepts/configuration/configmap.md:166]:
```suggestion
### Using ConfigMaps as files from a Pod
```

File: content/en/docs/concepts/configuration/configmap.md [hunk 2]
```diff
@@ -157,6 +157,91 @@ or {{< glossary_tooltip text="operators" term_id="operator-pattern" >}} that
...
+kind: ConfigMap
+metadata:
+  ...
+data:
+  ...
+immutable: true
+```
+
+{{< note >}}
+Once a Secret or ConfigMap is marked as immutable, it is _not_ possible to revert this change
```
Comment (sftim) [inline content/en/docs/concepts/configuration/configmap.md:240]: nit:
```suggestion
Once a ConfigMap or Secret is marked as immutable, it is _not_ possible to revert this change
```

Comment (wojtek-t) [reply content/en/docs/concepts/configuration/configmap.md]: done

File: content/en/docs/concepts/configuration/configmap.md (hunk 1 above)
Comment (wojtek-t) [reply content/en/docs/concepts/configuration/configmap.md]: done

**Pull request merged**